GANDLF/models/lightning_module.py

```diff
@@ -0,0 +1,27 @@
+import torch
+import lightning.pytorch as pl
+from GANDLF.models import get_model
+from GANDLF.optimizers import get_optimizer
+from GANDLF.schedulers import get_scheduler
+from GANDLF.losses.loss_calculators import LossCalculatorFactory
+from GANDLF.models.modelBase import ModelBase
+
+from copy import deepcopy
+
+
+class GandlfLightningModule(pl.LightningModule):
```

Comment:
We may move it, but I treat the model base as a base class for the architectures, and the lighting module is kind of orchestrator of not only architecture but also the training, inference loops etc. So I would basically say it is a higher level of abstraction, thus I put it in a separate file. Also this will never be subclassed I think.

Reply:
Ah okay, I understand, thank you!